modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/SetBasedExpiryMessageDeletionTask.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.andes.kernel;
+
+import com.google.common.util.concurrent.SettableFuture;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.andes.store.FailureObservingStoreManager;
+import org.wso2.andes.store.HealthAwareStore;
+import org.wso2.andes.store.StoreHealthListener;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * This task is responsible for delete ths expired messages captured from the message flusher
+ * Those captured messages accumulated in a set for a batch delete. This task is responsible to do
+ * that batch delete
+ */
+public class SetBasedExpiryMessageDeletionTask implements Runnable, StoreHealthListener {
+
+    /**
+     * hold the expired messages detected in the message flusher delivery rule check for batch delete
+     */
+    private static Set<DeliverableAndesMetadata> expiredMessageSet = new HashSet<>();
```

Comment:
We are having set in order to avoid the duplicates. I have the equals and hashcode method overridden in the Andes metadata class